### PR TITLE
Modulesync 2.7.0 and drop Puppet 4 support

### PR DESCRIPTION
**Pull request by pillarsdotnet**
- Exclude unnecessary nodesets via [`.sync.yml`](https://github.com/voxpupuli/puppet-openvmtools/pull/15/commits/fd49dd1d63cdb3771da969a12b68c26daf1ef92c)
- Run modulesync from https://github.com/voxpupuli/modulesync_config
- Rename `tests` to `examples`.
- Fix new rubocop errors.
- Tweak pre-commit config.
- Raise lower Puppet bound per @dhollinger.

### CODE REVIEW

File: spec/acceptance/nodesets/archlinux-2-x64.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
```
Comment (alexjfisher) [inline spec/acceptance/nodesets/archlinux-2-x64.yml]: These nodesets are probably not required.  See https://github.com/voxpupuli/puppet-redis/pull/299#discussion_r264323948

File: metadata.json [hunk 2]
```diff
@@ -103,7 +103,7 @@
   "requirements": [
     {
       "name": "puppet",
-      "version_requirement": ">=4.10.0 <7.0.0"
+      "version_requirement": ">=5.0.0 <7.0.0"
```
Comment (alexjfisher) [inline metadata.json]: I think the minimum version we actually support is higher than this.

```suggestion
      "version_requirement": ">= 5.5.8 < 7.0.0"
```

File: spec/acceptance/nodesets/archlinux-2-x64.yml (hunk 1 above)
Comment (pillarsdotnet) [reply spec/acceptance/nodesets/archlinux-2-x64.yml]: So I should remove that file?

Comment (ekohl) [reply spec/acceptance/nodesets/archlinux-2-x64.yml]: You should also mark it as delete in modulesync. From the same linked repo: https://github.com/voxpupuli/puppet-redis/blob/d0ee587d0d0327ef85d5701fbd6c9d9f33569fbe/.sync.yml#L17-L30

Comment (pillarsdotnet) [reply spec/acceptance/nodesets/archlinux-2-x64.yml]: Read the comment and followed the example therein.

Comment (pillarsdotnet) [reply spec/acceptance/nodesets/archlinux-2-x64.yml]: Okay; I'll add the other lines, too.

**Comment by alexjfisher** [comment]
Thanks @pillarsdotnet !

**Pull request merged**